src/averageAges.js

```diff
@@ -1,25 +1,23 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
+/*
  *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter((person) => person.sex === 'm');
+  const allMen = century
+    ? men.filter(person => Math.ceil(person.died / 100) === century)
+    : men;
+
+  const averageMen = allMen.reduce((sum, year) => (year.died - year.born)
```

Comment:
Same here, you calculate average more than once down the line, use function for those calculations.